2024-03-17 17:35:51.789855: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below 5.51GiB (5920943308 bytes) by rematerialization; only reduced to 52.00GiB (55834574864 bytes), down from 52.00GiB (55834574864 bytes) originally
2024-03-17 17:35:51.789833: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below 5.51GiB (5920943308 bytes) by rematerialization; only reduced to 52.00GiB (55834574864 bytes), down from 52.00GiB (55834574864 bytes) originally
2024-03-17 17:35:51.821809: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below 5.51GiB (5920943308 bytes) by rematerialization; only reduced to 52.00GiB (55834574864 bytes), down from 52.00GiB (55834574864 bytes) originally
2024-03-17 17:35:51.821812: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below 5.51GiB (5920943308 bytes) by rematerialization; only reduced to 52.00GiB (55834574864 bytes), down from 52.00GiB (55834574864 bytes) originally
2024-03-17 17:36:01.857519: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_2_bfc) ran out of memory trying to allocate 16.00GiB (rounded to 17179869440)requested by op 
2024-03-17 17:36:01.857620: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_3_bfc) ran out of memory trying to allocate 16.00GiB (rounded to 17179869440)requested by op 
2024-03-17 17:36:01.857750: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*****************************************************************************_______________
2024-03-17 17:36:01.857828: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*****************************************************************************_______________
E0317 17:36:01.857854 3685134 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 17179869312 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   16.00GiB
     preallocated temp allocation:   16.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:   34.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 2:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: fft
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f64[256,1024,1024]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: (c128[256,1024,1024], c128[1024,1024,1024])
		==========================


E0317 17:36:01.857891 3685135 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 17179869312 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   16.00GiB
     preallocated temp allocation:   16.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:   34.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 2:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: fft
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f64[256,1024,1024]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: (c128[256,1024,1024], c128[1024,1024,1024])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 110, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 57, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 17179869312 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   16.00GiB
     preallocated temp allocation:   16.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:   34.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 2:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: fft
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f64[256,1024,1024]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: (c128[256,1024,1024], c128[1024,1024,1024])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 110, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 57, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 17179869312 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   16.00GiB
     preallocated temp allocation:   16.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:   34.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 2:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: fft
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f64[256,1024,1024]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: (c128[256,1024,1024], c128[1024,1024,1024])
		==========================


2024-03-17 17:36:01.888259: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_1_bfc) ran out of memory trying to allocate 16.00GiB (rounded to 17179869440)requested by op 
2024-03-17 17:36:01.888482: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*****************************************************************************_______________
E0317 17:36:01.888594 3685133 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 17179869312 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   16.00GiB
     preallocated temp allocation:   16.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:   34.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 2:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: fft
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f64[256,1024,1024]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: (c128[256,1024,1024], c128[1024,1024,1024])
		==========================


2024-03-17 17:36:01.888618: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_0_bfc) ran out of memory trying to allocate 16.00GiB (rounded to 17179869440)requested by op 
2024-03-17 17:36:01.888832: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*****************************************************************************_______________
E0317 17:36:01.888964 3685132 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 17179869312 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   16.00GiB
     preallocated temp allocation:   16.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:   34.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 2:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: fft
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f64[256,1024,1024]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: (c128[256,1024,1024], c128[1024,1024,1024])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 110, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 57, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 17179869312 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   16.00GiB
     preallocated temp allocation:   16.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:   34.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 2:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: fft
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f64[256,1024,1024]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: (c128[256,1024,1024], c128[1024,1024,1024])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 110, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 57, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 17179869312 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   16.00GiB
     preallocated temp allocation:   16.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:   34.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 2:
		Size: 16.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: fft
		Shape: c128[1024,1024,1024]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f64[256,1024,1024]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(1024, 1024, 1024)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=44
		XLA Label: all-gather-start
		Shape: (c128[256,1024,1024], c128[1024,1024,1024])
		==========================


srun: error: r6i6n0: tasks 0-2: Exited with exit code 1
srun: Terminating StepId=1125458.57
slurmstepd: error: *** STEP 1125458.57 ON r6i6n0 CANCELLED AT 2024-03-17T17:36:02 ***
srun: error: r6i6n0: task 3: Exited with exit code 1
